eta/core/datasets.py

```diff
@@ -1797,6 +1820,14 @@ def transform(self, src):
         for ki in keep_idxs:
             src.add(old_records[record_idxs[ki]])
 
+    def _validate(self):
+        if all([self.attr_name is None,
```

Comment:
either `attr_name` or `labels_schema` must be provided. `object_label` is used with `attr_name` if provided, but is not a sufficient input on its own.

Reply:
Yes, that is verified [here](https://github.com/voxel51/eta/blob/7f1cd8c483bff7adb20ce56e8472c15486310773/eta/core/datasets.py#L1909-L1912).  Are you suggesting we move that check into the `_validate()` function?

Reply:
Well, I think you could get rid of that second check, and do this:

```
if ((self.attr_name is not None and self.labels_schema is not None)
                or (self.attr_name is None and self.labels_schema is None)):
            raise ValueError(
                "One and only one of the following must be specified: "
                "attr_name, labels_schema")

        if self.object_label is not None and self.attr_name is None:
            raise ValueError(
                "attribute_name must be specified if object_label is "
                "specified")
```

Reply:
Ok, I changed things around a bit, is this clearer now?

Reply:
yep!